github/git_refs.go

```diff
@@ -58,13 +60,56 @@ func (s *GitService) GetRef(ctx context.Context, owner string, repo string, ref
 
 	r := new(Reference)
 	resp, err := s.client.Do(ctx, req, r)
-	if err != nil {
+	if _, ok := err.(*json.UnmarshalTypeError); ok {
+		// Multiple refs, means there wasn't an exact match.
+		return nil, resp, nil
+	} else if err != nil {
 		return nil, resp, err
 	}
 
 	return r, resp, nil
 }
 
+// GetRefs fetches a slice of Reference objects for a given Git ref.
+// If there is an exact match, only that ref is returned.
+// If there is no exact match, GitHub returns all refs that start with ref.
```

Comment:
From my testing, when there are multiple matches returned, there's always at least 1 entry, otherwise it's a 404 not found error.

I think we can take advantage of that to make the API friendlier. We can make a guarantee that if returned `error` is `nil`, then there's at least one match, making it safe to access `refs[0]`.

I think doing that would make it easier for people to use `GetRefs`. What are your thoughts on this?

I'm imagining something like:

```
// If there is an exact match, only that ref is returned.
// If there is no exact match, GitHub returns all refs that start with ref.
// If returned error is nil, there will be at least 1 ref returned.
```

Or is this not worth doing?